src/core/player.js

```diff
@@ -218,6 +218,20 @@ DM.provide('Player',
         }
     },
 
+    _decodePostMessage: function(rawMessage)
+    {
+      if (rawMessage.substring(0, 1) === '{') {
```

Comment:
When we receive a `message`, we don't know yet to which player is it addressed. We need to decode the payload first, to get the player id. And the message mode depends on the player, example:

```js
var player1 = DM.player(document.getElementById("player"), {params: {apimode: 'json'}});
var player2 = DM.player(document.getElementById("player"), {params: {apimode: 'json'}});
```

So, checking the payload itself to decide if we json-decode or QS-decode is the only easy method I see (without refactoring the whole `_installHandlers` helper).